### PR TITLE
Added Focusnodes in form sample

### PR DESCRIPTION
**Pull request by Alabhya268**
Added FocusNodes to improve user experience in form_app.

### CODE REVIEW

**Comment by Alabhya268** [comment]
@RedBrogdon  @johnpryan , Please review this PR , Let me if anything needs to be changed.

File: form_app/lib/src/autofill.dart [hunk 1]
```diff
@@ -14,6 +14,32 @@ class AutofillDemo extends StatefulWidget {
 class _AutofillDemoState extends State<AutofillDemo> {
   final _formKey = GlobalKey<FormState>();
 
+  FocusNode _lastname, _email, _telephone, _streetAddresss, _postalCode, _country, _countrycode;
```
Comment (RedBrogdon) [inline form_app/lib/src/autofill.dart:17]: nit: We generally use a max line length of 80 in this repo, and typically break fields out onto their own lines, even if the types are the same:

```dart
FocusNode _lastName;
FocusNode _mail;
... etc. ...
```

File: form_app/lib/src/autofill.dart [hunk 2]
```diff
@@ -31,13 +57,21 @@ class _AutofillDemoState extends State<AutofillDemo> {
                   ...[
                     Text('This sample demonstrates autofill. '),
                     TextFormField(
+                      onFieldSubmitted: (vaue) {
+                        _lastname.requestFocus();
```
Comment (RedBrogdon) [inline form_app/lib/src/autofill.dart:61]: If we're adding focus nodes to the other fields, it's probably a good idea to add one to this field as well, in order to complete the set. I wouldn't be surprised if it mattered for tab-order when building for the web, for example.

File: form_app/lib/src/sign_in_http.dart [hunk 3]
```diff
@@ -54,16 +68,19 @@ class _SignInHttpDemoState extends State<SignInHttpDemo> {
               children: [
                 ...[
                   TextFormField(
+                    textInputAction: TextInputAction.next,
```
Comment (RedBrogdon) [inline form_app/lib/src/sign_in_http.dart:71]: Similar to the above, I'd recommend adding focus nodes to all the fields in this and the other forms.

**Review by RedBrogdon — CHANGES_REQUESTED** [review]
I added a few nits, then realized @johnpryan is probably better suited to review this. 😄

**Comment by Alabhya268** [comment]
@RedBrogdon , I made the necessary changes  : )

**Review by RedBrogdon — APPROVED** [review]
LGTM

File: form_app/lib/src/autofill.dart [hunk 4]
```diff
@@ -31,13 +66,23 @@ class _AutofillDemoState extends State<AutofillDemo> {
...
                         hintText: 'Jane',
                         labelText: 'First Name',
                       ),
                       autofillHints: [AutofillHints.givenName],
                     ),
                     TextFormField(
+                      focusNode: _lastName,
+                      textInputAction: TextInputAction.next,
+                      onFieldSubmitted: (value) {
+                        _email.requestFocus();
```
Comment (johnpryan) [inline form_app/lib/src/autofill.dart:84]: Manually focusing the next field is not required when using `TextInputAction.next`, so the FocusNodes can be removed.

File: form_app/lib/src/autofill.dart [hunk 5]
```diff
@@ -31,13 +66,23 @@ class _AutofillDemoState extends State<AutofillDemo> {
                   ...[
                     Text('This sample demonstrates autofill. '),
                     TextFormField(
+                      autofocus: true,
+                      focusNode: _firstName,
```
Comment (johnpryan) [inline form_app/lib/src/autofill.dart:70]: Add `textInputAction: TextInputAction.next` to this field too.

**Review by johnpryan — CHANGES_REQUESTED** [review]

File: form_app/lib/src/autofill.dart (hunk 4 above)
Comment (Alabhya268) [reply form_app/lib/src/autofill.dart]: I didn't knew it, I've now removed unnecessary focusnodes

**Comment by Alabhya268** [comment]
@johnpryan ,Please review this PR again.

**Review by johnpryan — APPROVED** [review]

**Pull request merged**